src/java/org/apache/cassandra/cql3/statements/schema/CreateKeyspaceStatement.java

```diff
@@ -87,10 +83,6 @@ public Keyspaces apply(Keyspaces schema)
         keyspace.params.validate(keyspaceName);
         Keyspaces keyspaces = schema.withAddedOrUpdated(keyspace);
 
-        int newNumWarnings = ClientWarn.instance.numWarnings();
-        if (newNumWarnings > previousNumWarnings)
-            clientWarnings.addAll(ClientWarn.instance.getWarnings().subList(previousNumWarnings, newNumWarnings));
-
```

Comment:
I don't think we are ignoring anything. AFAIR client warning works in the way that once we start processing a query, client warnings state is initialized and since that time whenever we do `ClientWarn.instance.warn` it is added to the client warnings internal state which is for this particular client request. The workaround here was because state of client warnings was not passed from the initial thread which processed the request to the subtasks processed in different threads (in this case, the task was executed in migration stage).

Reply:
btw. I can add an initial warning by byteman, before we get into the apply method and check if it is not cleared

Reply:
We're not ignoring warnings, but adding just the _newly generated ones_. Otherwise they would accumulate and duplicate depending on how state and multi-threading played at each particular scenario.

Reply:
Indeed that was a workaround to avoid accumulating warnings from previous queries. With the proposed changes we still get the proper warnings without needing that code to remove repetitions. This can be verified for example by running these two queries in a single node cluster:
```
CREATE KEYSPACE k1 WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 2}; # warns
CREATE KEYSPACE k2 WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1}; # doesn't warn
```
We also have some tests to verify that this works as expected.

Reply:
> avoid accumulating warnings from previous queries

might not understand what you are saying, but if ClientWarn contains state from a different request, then that is a bug (which I would assume is fixed by this PR?).

> We also have some tests to verify that this works as expected.

so if they pass that means this PR fixes the root cause?  Works for me =)

Reply:
It is not from other queries but from the same. You want

> Your warning 1
> Your warning 2

instead of

> Your warning 1
> Your warning 1
> Your warning 2

Makes sense? But I'm terrible at explaining things lol